### PR TITLE
Move column specific formatting into a new dropdown section

### PR DESCRIPTION
**Pull request by emilyhuxng**
Fixes [#994](https://github.com/deephaven/web-client-ui/issues/994)

I removed the code returning column specific formatting and any code that becomes unnecessary afterwards in FormattingSectionContent. Then in ColumnSpecificSectionContent, I took the original code and did the same thing but removed code related to Timezone formatting.

Also renamed the sections.

### CODE REVIEW

**Comment by emilyhuxng** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by mofojed — CHANGES_REQUESTED** [review]
Looks great!
Minor change - in the PR description, change it to say "Fixes #994" instead of just "#994". By adding the "Fixes" keyword, it will close the ticket #994 when this is merged.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+    const { id, isNewRule, ...rest } = item;
+    return rest;
+  }
+
+  static isFormatRuleValidForSave(rule: FormatterItem): boolean {
+    return (
+      ColumnSpecificSectionContent.isValidColumnName(rule.columnName) &&
+      ColumnSpecificSectionContent.isValidFormat(rule.columnType, rule.format)
+    );
+  }
```
Comment (mofojed) [inline packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx:186]: I think all of these static methods are also used in `FormattingSectionContent`. You should refactor these into their own utils file, e.g. `SettingsUtils.ts`. See `FormatterUtils.ts` for an example of a utils file (exporting a bunch of functions).

**Review by mofojed — CHANGES_REQUESTED** [review]

File: packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx [hunk 2]
```diff
@@ -35,31 +33,11 @@ import {
...
-};
-
-function isFormatStringFormat(
-  format: Partial<TableColumnFormat>
-): format is Pick<TableColumnFormat, 'formatString'> {
-  return (
-    (format as Pick<TableColumnFormat, 'formatString'>).formatString != null
-  );
-}
+import SettingsUtils from './SettingsUtils';
```
Comment (mofojed) [inline packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx:39]: You could also import just the functions you use directly, so you don't need to call `SettingsUtils.*` each time, e.g. 

```
import { focusFirstInputInContainer } from './SettingsUtils';
...
  componentDidMount(): void {
    focusFirstInputInContainer(this.containerRef.current);
  }
```

File: packages/code-studio/src/settings/SettingsUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        {formatter.getFormattedString(
+          timestamp,
+          TableUtils.dataType.DATETIME,
+          '',
+          format
+        )}
+      </option>
+    );
+  });
+}
```
Comment (mofojed) [inline packages/code-studio/src/settings/SettingsUtils.tsx:185]: These static functions are essentially just functional React components, they could be put into their own files, e.g. `TimeZoneOptions.tsx`, `ColumnTypeOptions.tsx`, and `DateTimeOptions.tsx`. Not a big deal though.

**Review by mofojed — APPROVED** [review]
A couple minor style suggestions, looks great!

File: packages/code-studio/src/settings/ColumnTypeOptions.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React, { ReactElement } from 'react';
+import { TableUtils } from '@deephaven/jsapi-utils';
+
+export default function ColumnTypeOptions(): ReactElement {
+  const columnTypesArray = [
+    { value: TableUtils.dataType.DATETIME, label: 'DateTime' },
+    { value: TableUtils.dataType.DECIMAL, label: 'Decimal' },
+    { value: TableUtils.dataType.INT, label: 'Integer' },
+  ];
```
Comment (mofojed) [inline packages/code-studio/src/settings/ColumnTypeOptions.tsx:9]: This const could be initialized outside of the function.

File: packages/code-studio/src/settings/DateTimeOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+} from '@deephaven/jsapi-utils';
+
+export default function renderDateTimeOptions(
+  timestamp: Date,
+  timeZone: string,
+  showTimeZone: boolean,
+  showTSeparator: boolean,
+  isGlobalOptions: boolean,
+  legacyGlobalFormat?: string
+): ReactElement[] {
```
Comment (mofojed) [inline packages/code-studio/src/settings/DateTimeOptions.tsx:15]: Should make this a functional component, e.g.

```
interface DateTimeOptionsProps {
  timestamp: Date;
  timeZone: string;
  showTimeZone: boolean;
  showTSeparator: boolean;
  isGlobalOptions: boolean;
  legacyGlobalFormat?: string;
}

export default function DateTimeOptions(props: DateTimeOptionsProps) {
  ...
}
```

File: packages/code-studio/src/settings/TimeZoneOptions.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { TimeUtils } from '@deephaven/utils';
+
+export default function renderTimeZoneOptions(): JSX.Element[] {
```
Comment (mofojed) [inline packages/code-studio/src/settings/TimeZoneOptions.tsx:4]: Make this a function component, e.g.
```
export default function TimeZoneOptions(): JSX.Element {
  ...
}
```

**Review by mofojed — CHANGES_REQUESTED** [review]

**Review by mofojed — APPROVED** [review]
Looking good. How are the tests coming?

**Comment by emilyhuxng** [comment]
> Looking good. How are the tests coming?

I have 100% code coverage but branch coverage is at 85% so I am working on that.

**Pull request merged**